python/ray/air/_internal/checkpoint_manager.py

```diff
@@ -320,21 +323,22 @@ def register_checkpoint(self, checkpoint: _TrackedCheckpoint):
         Args:
             checkpoint: Tracked checkpoint object to add to bookkeeping.
         """
-        checkpoint.id = checkpoint.id or self._latest_checkpoint_id
+        for checkpoint in checkpoints:
+            checkpoint.id = checkpoint.id or self._latest_checkpoint_id
 
-        if checkpoint.storage_mode == CheckpointStorage.MEMORY:
-            self._replace_latest_memory_checkpoint(checkpoint)
+            if checkpoint.storage_mode == CheckpointStorage.MEMORY:
+                self._replace_latest_memory_checkpoint(checkpoint)
 
-            if self._persist_memory_checkpoints:
-                persisted_checkpoint = copy.copy(checkpoint)
-                persisted_checkpoint.storage_mode = CheckpointStorage.PERSISTENT
+                if self._persist_memory_checkpoints:
+                    persisted_checkpoint = copy.copy(checkpoint)
+                    persisted_checkpoint.storage_mode = CheckpointStorage.PERSISTENT
+                else:
+                    persisted_checkpoint = None
             else:
-                persisted_checkpoint = None
-        else:
-            persisted_checkpoint = checkpoint
+                persisted_checkpoint = checkpoint
 
-        if persisted_checkpoint and self._checkpoint_strategy.num_to_keep != 0:
-            self._process_persistent_checkpoint(persisted_checkpoint)
+            if persisted_checkpoint and self._checkpoint_strategy.num_to_keep != 0:
+                self._process_persistent_checkpoint(persisted_checkpoint)
```

Comment:
Should this pass the `next_checkpoint_path`?

Reply:
we need to pass the checkpoint here.
I am actually not changing the logics here. just wrapping it inside a for loop.

Reply:
Ah I see. I guess I was confused because I thought `self._get_next_checkpoint_path()` increases the iteration, but it doesn't.

Do we still need the `next_checkpoint_path` argument for `_process_persistent_checkpoint()` then? Looks like it's not used